### PR TITLE
config: fix specified settings

### PR DESCRIPTION
**Pull request by runcom**
I've noticed the cri-o RPM generates the configuration using this
snippet:

./bin/crio ... --storage-driver=overlay
--storage-opt="overlay.override_kernel_check=1" config ...

However, the storage options aren't actually set because they are
commented out in the config stub we use to generate the config file.
This has worked till now because /etc/containers/storage.conf contains
valid settings used implicitely by cri-o but we should not rely on that
since we're actually generating the configuration ourselves.

This commit add checks around the config generation to make sure the
correct options are picked and printed.
This commit also de-comment ulimits settings as that's the correct
thing to do since the empty list is the same as commenting it out and
also, it's needed if someone specifies a config with ulimits.

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by giuseppe** [comment]
LGTM

File: cmd/crio/config.go [hunk 1]
```diff
@@ -83,8 +87,8 @@ stream_tls_ca = "{{ .StreamTLSCA }}"
 # "<ulimit name>=<soft limit>:<hard limit>", for example:
 # "nofile=1024:2048"
 # If nothing is set here, settings will be inherited from the CRI-O daemon
-#default_ulimits = [
-{{ range $ulimit := .DefaultUlimits }}{{ printf "#\t%q,\n" $ulimit }}{{ end }}#]
+default_ulimits = [
+{{ range $ulimit := .DefaultUlimits }}{{ printf "\t%q,\n" $ulimit }}{{ end }}]
```
Comment (TomSweeneyRedHat) [inline cmd/crio/config.go]: I'm assuming it's ok for the last option to have a comma at the end of it and no option afterwards?

Comment (runcom) [reply cmd/crio/config.go]: yes

**Comment by runcom** [comment]
@sboeuf hey, looks like kata jobs disappeared, do you know what's going on?

**Comment by runcom** [comment]
cronjob e2e failures should be fixed by https://github.com/kubernetes/kubernetes/pull/70206
The other failure in nfs is being investigated

**Comment by rhatdan** [comment]
As stated in the back port PR.  I am against this change.  CRI-O By default should use storage.conf and have its storage options commented out.  Admins should be forced to change the Configuraiton in crio.conf only if they want to differ then the configuration that podman/buildah use.  If out of the box crio.conf has a different configuration, then the admin will be forced to maintain TWO configuration files for storage if they want to allow sharing of images between CRI-O, Buildah and Podman.

**Comment by runcom** [comment]
> As stated in the back port PR. I am against this change. CRI-O By default should use storage.conf and have its storage options commented out. Admins should be forced to change the Configuraiton in crio.conf only if they want to differ then the configuration that podman/buildah use. If out of the box crio.conf has a different configuration, then the admin will be forced to maintain TWO configuration files for storage if they want to allow sharing of images between CRI-O, Buildah and Podman.

Dan, ptal at my comment in the other PR. This PR doesn't affect the whole crio behavior wrt storage options. This is actually _fixing_ the scenario where someone specifies storage options to create a config file.

**Comment by sboeuf** [comment]
@chavafg any idea why the `kata-jenkins` job vanished?

**Comment by chavafg** [comment]
not sure, our jenkins seems that has not received requests since Oct 18th. 
Any idea @stevekuznetsov ?

**Comment by runcom** [comment]
kata's back ;)

**Comment by sboeuf** [comment]
Nice!
/cc @chavafg

**Comment by chavafg** [comment]
cool :) but now it will be failing by #1883

**Comment by runcom** [comment]
> cool :) but now it will be failing by #1883

we'll fix that no worry :) thanks for reporting it!

**Comment by runcom** [comment]
it's probably a regression introduced while the kata ci was not able to contact our CI, we'll chase it down!

**Comment by runcom** [comment]
to get back on this, I strongly feel we need to fix our code with this PR since, while it's true we want to honor /etc/containers/storage.conf, and we continue to do so with this PR as well, we need to let _upstream_ and _users_ do what they want since we allow that. The rule was we'll use default config files across tools till someone overrides something directly in these tools (in this example, it's CRI-O). So, this PR doesn't change the behavior of honoring the storage.conf file, but it provides a working and really user-definied configuration file generated by the `crio config` subcommand.

**Comment by rhatdan** [comment]
@runcom Still working on this?

**Comment by umohnani8** [comment]
Changes LGTM.
@mrunalp @rhatdan tests are green, PTAL.

**Comment by rhatdan** [comment]
@umohnani8 Could you finish this one up for @runcom if it is still needed?

**Comment by umohnani8** [comment]
@runcom do we still need this? We have templates for storage.conf and crio.conf that are being put down by MCO now.

**Comment by rhatdan** [comment]
I am closing since this is ancient and has had no feeback in many months, reopen if people want to make progress on it.

**Pull request closed without merging**